### ISSUE TITLE
[Bug]: Failing Docker build is ignoring `build_flag_templates` flags

### ISSUE THREAD

**Issue by mieubrisse** [goreleaser/goreleaser#2438]
### What happened?

I wanted to copy some extra files into one of my Docker images. I added the `extra_files` key to the appropriate Docker build in my `goreleaser.yml`:
```
  - id: internal-testsuite
    dockerfile: internal_testsuite/Dockerfile
    image_templates:
      # TODO EXTRACT DOCKER ORG & REPO NAME INTO ENVVARS
      - "kurtosistech/kurtosis-core_internal-testsuite:{{ .Tag }}"
    goos: linux
    goarch: amd64
    extra_files:
      - internal_testsuite/static_files
```
However, my Dockerfile:
```
FROM alpine:3.12

WORKDIR /run

# Copy the code into the container
COPY internal-testsuite .

# Copy local static files inside the testsuite image
COPY static_files /static-files

CMD ./internal-testsuite
```
 was failing on the COPY step with `#9 ERROR: "/static_files" not found: not found`.
I then added:
```
RUN find .
```
to debug, but the step was getting cached:
```
         ⨯ #8 [4/5] RUN find .       cmd=[docker build] cwd=dist/goreleaserdocker650176556
         ⨯ #8 sha256:846e53aa2b59688c59f89daa5b37e62519eaabefb17196452c654b7da6368046 cmd=[docker build] cwd=dist/goreleaserdocker650176556
         ⨯ #8 CACHED                 cmd=[docker build] cwd=dist/goreleaserdocker650176556
```
I then added the following to my Docker build config in my `goreleaser.yml`:
```
    build_flag_templates:
      - "--no-cache"
```
but the step still kept getting cached.

I eventually figured out, by inspecting the `dist/goreleaser*` directories, that my COPY command needed to be `COPY internal_testsuite/static_files /static-files`, but the `--no-cache` not getting applied seems to still be an issue.

I've attached my goreleaser.yml and the output of running `goreleaser release --debug --rm-dist --snapshot`.

[my-archive.zip](https://github.com/goreleaser/goreleaser/files/7071282/my-archive.zip)

### How can we reproduce this?

Try adding the `--no-cache` build flag template to a failing Docker build.

### goreleaser version

```bash
goreleaser version 0.176.0
commit: dd5ccf7170741e0303e90e81faf9073d03ac8f43
built by: homebrew

https://goreleaser.com
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
If you run with --debug, what's the Docker args goreleaser is using?

**Comment by mieubrisse** [comment]
Hey @caarlos0 thanks for the response! That zip actually contains the `--debug` output - AFAICT, it's just regular Docker build:
```
         • running                   cmd=[docker build] cwd=dist/goreleaserdocker988340286
```
even though the extra flags are correctly getting detected from the config file:
```
.... [--no-cache --label=test=TESTVALUE] [] false }] [] [] [] [] [] {{[]}  true}
```

**Comment by caarlos0** [comment]
oh, it doesn't print the args... just fixed it on 1052ce17 so it does now.

and it its passing down the flags as expected 🤔 


```
         • running                   args=[build . --load -t goreleaser/goreleaser:v0.176.0-amd64 -t ghcr.io/goreleaser/goreleaser:v0.176.0-amd64 --pull --label=org.opencontainers.image.created=2021-09-01T02:31:55Z --label=org.opencontainers.image.name=goreleaser --label=org.opencontainers.image.revision=c08bb7f24c693850b527840135f37364bfdb7a84 --label=org.opencontainers.image.version=0.176.1-next --label=org.opencontainers.image.source=git@github.com:goreleaser/goreleaser.git --platform=linux/amd64] cmd=[docker buildx] cwd=dist/goreleaserdocker4241676158
```

if `--no-cache` is not having any effect, it might be something on docker itself or in your dockerfile...

**Comment by caarlos0** [comment]
closing as I cannot reproduce... feel free to comment if you find more info
